### PR TITLE
Remove localhost from code

### PR DESCRIPTION
**Pull request by lazzarello**
Small changes to the next.js setup configuration and documentation to allow the frontend and backend to be de-coupled without any change of functionality. This allows for the backend transcoder to be taken offline without the frontend going down with it.

Right now the frontend will display a message that "owncast is offline" which is true. This design could be improved.

### CODE REVIEW

**Comment by gabek** [comment]
It seems you checked in far more files than you intended to. Could you fix that up so it can be seen what was meant to change?

**Comment by lazzarello** [comment]
I ran a static build from `./build/web/bundleWeb.sh` not sure what to fix since this would change the static site, no?

**Comment by gabek** [comment]
Don't check in any files under `static/web`. Those are generated.

**Comment by lazzarello** [comment]
does that mean it should be under the .gitignore?

**Comment by gabek** [comment]
CI will run a build and check in whatever is needed whenver changes to the web project get made. They shouldn't be ignored because when people clone the project they are required to run it.

**Comment by gabek** [comment]
In general no human should ever need to run `./build/web/bundleWeb.sh`. If you do, and check in the changes, you get what you see above, a hundred conflicts.

**Comment by lazzarello** [comment]
only 25 but I see your point. I suspected that was how a "new release" was built. I'll just replace them with the contents of HEAD at the develop branch.

**Comment by lazzarello** [comment]
not sure where that `static/web/_next` modification came from. My local repo said I modified the following files after copying the whole static dir from develop.

```
        modified:   static/web/404.html
        modified:   static/web/404/index.html
        modified:   static/web/admin/access-tokens/index.html
        modified:   static/web/admin/actions/index.html
        modified:   static/web/admin/chat/emojis/index.html
        modified:   static/web/admin/chat/messages/index.html
        modified:   static/web/admin/chat/users/index.html
        modified:   static/web/admin/config-chat/index.html
        modified:   static/web/admin/config-federation/index.html
        modified:   static/web/admin/config-notify/index.html
        modified:   static/web/admin/config-social-items/index.html
        modified:   static/web/admin/config-video/index.html
        modified:   static/web/admin/config/general/index.html
        modified:   static/web/admin/config/server/index.html
        modified:   static/web/admin/federation/actions/index.html
        modified:   static/web/admin/federation/followers/index.html
        modified:   static/web/admin/hardware-info/index.html
        modified:   static/web/admin/help/index.html
        modified:   static/web/admin/index.html
        modified:   static/web/admin/logs/index.html
        modified:   static/web/admin/stream-health/index.html
        modified:   static/web/admin/upgrade/index.html
        modified:   static/web/admin/viewer-info/index.html
        modified:   static/web/admin/webhooks/index.html
        modified:   static/web/embed/chat/readonly/index.html
        modified:   static/web/embed/chat/readwrite/index.html
        modified:   static/web/embed/video/index.html
        modified:   static/web/index.html
        modified:   static/web/sw.js
```

The computer says there are no more merge conflicts!

**Comment by gabek** [comment]
Did you mean to check in 10 files? That seems still more than what you intended to change.

**Comment by lazzarello** [comment]
I think I fixed it. Can you check?

**Comment by jprjr** [comment]
So to recap real quick: the core issue this is trying to solve - since owncast serves the pre-compiled front-end code, bringing the owncast process down results in the entire web app being unavailable.

I haven't tested this directly, but I think doing a split setup would break server-side hydration which is really useful for sharing links to the stream.

File: web/components/stores/ClientConfigStore.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ let serverStatusRefreshPoll: ReturnType<typeof setInterval>;
 let hasBeenModeratorNotified = false;
 let hasWebsocketDisconnected = false;
 
-const serverConnectivityError = `Cannot connect to the Owncast service. Please check your internet connection and verify this Owncast server is running.`;
+const serverConnectivityError = `Cannot connect to the backend service. Please check your internet connection and verify this Owncast server is running.`;
```
Comment (gabek) [inline web/components/stores/ClientConfigStore.tsx:41]: This was specifically called out to be the "Owncast service" to be explicit about the error. To a end user, a viewer watching the stream, to tell them "the backend is down" doesn't mean anything to them.

File: test/automated/browser/run.sh [hunk 2]
```diff
@@ -32,6 +32,7 @@ fi
 # Install the web test framework
 if [ -z "$SKIP_BUILD" ]; then
 	echo "Installing test dependencies..."
+	export PUPPETEER_SKIP_DOWNLOAD=true
```
Comment (gabek) [inline test/automated/browser/run.sh:35]: What is this?

**Comment by gabek** [comment]
> So to recap real quick: the core issue this is trying to solve - since owncast serves the pre-compiled front-end code, bringing the owncast process down results in the entire web app being unavailable.
> 
> I haven't tested this directly, but I think doing a split setup would break server-side hydration which is really useful for sharing links to the stream.

The devil is in the details with all of this. I'll mention what I've been saying in the Owncast chat, I'm not interested in supporting a split frontend and backend setup, and there isn't any discussion about _how_ that were to work in the first place. I'm assuming it would mean that I'd have to ship separate frontend and backend binaries, and that's not something I'm willing to do.

Fixing the hardcoded `localhost`s to use the environment variable, however, is a good change.

**Pull request closed without merging**